### ISSUE TITLE
FATA[0001] error on sending allocate request: ConnectionRead: timeout occurred

### ISSUE THREAD

**Issue by Himself132** [firefart/stunner#5]
Hello, super cool tool. Had an opportunity for a security assessment to check out a STUN server but am running into some issues with the tool.

nmap and nessus are both reporting listeners on several STUN servers on port 3478 via UDP and are grabbing banners.

No matter what flags I pass I see the following (even if i specify --protocol udp). This happens on all servers I'm testing.  If also choosing tls i just get a similar error about a timeout.

DEBU[0000] Sending
Header:
        Message Type: Allocate(03) Request(00)
        Message Length: 0
        Message Transaction ID: 66736c6e7a46425a61454747
Attributes
        REQUESTED-TRANSPORT: TCP 
FATA[0001] error on sending allocate request: ConnectionRead: timeout occurred 

Wireshark for your tool is showing Allocate Request TCP (REQUESTED-TRANSPORT TCP) but it is in UDP packets.  nmap sends a binding request. not sure if this is a problem with my limited knowledge of STUN or it's just that the servers I'm scanning are hardened and won't allow any of the attacks your tool does.

**Comment by firefart** [comment]
Hi,
you can try to increase the timeout. The error message occirs if a reply does not come back in the defined timeout span.

The BindRequest ist only done on UDP based allocations (the protocol from the cli parameter is just the protocol to the turn server itself, not the protocol the turn server uses to connect to backends). So it might also be the case that the TCP protocol via TURN is not activated. You can try the `brute-transports` command to see if TCP is supported or not.

To verify the connection to the server you can use the `info` command and later on the `range-scan` to detect the valid IP ranges but I guess it's just a too short timeout (default is 1 second).

So long story short:
`--timeout 5s` :)

**Comment by Himself132** [comment]
Hello, the issue is actually when I am using the `info` command and I have tried the timeout values being higher.  nmap and nessus sees the service listening and an internal IP address exposed.  if you use wireshark you can see nmap sending a binding request but your tool is just sending an allocation request for an `info` request and it shows TCP.  if I run the tool with --protocol udp it still shows the TCP in the `info` test

**Comment by firefart** [comment]
The you are dealing most likely with a STUN only server without TURN functionality. I will try to detect that in the info command too

**Comment by firefart** [comment]
The `info` command now prints out more information about the server like supported protocols. Thanks for the suggestion!

**Comment by Himself132** [comment]
So if i install the new one I should have more information? thank you! Also is there a way for me to know for sure if they don't have TURN functionality?

**Comment by firefart** [comment]
The new `info` command will show you if it support STUN, TURN and/or TURN via TCP. If this only yields the STUN protocol than it's pretty useless as the STUN protocol can only reveal your public facing IP
